src/java/org/apache/cassandra/db/Keyspace.java

```diff
@@ -337,7 +337,19 @@ private Keyspace(String keyspaceName, boolean loadSSTables)
         for (TableMetadata cfm : metadata.tablesAndViews())
         {
             logger.trace("Initializing {}.{}", getName(), cfm.name);
-            initCf(Schema.instance.getTableMetadataRef(cfm.id), loadSSTables);
+            TableMetadataRef tableMetadataRef = Schema.instance.getTableMetadataRef(cfm.id);
+            
+            if (tableMetadataRef == null)
+            {
+                logger.info("Failed to initialize {}.{}, as no table metadata was available. This " +
+                            "has likely occurred because the keyspace metadata for {} has not yet " +
+                            "been updated to reflect the table being dropped.", 
+                            getName(), cfm.name, getName());
```

Comment:
The one thing that did cross my mind is whether or not simply logging here wouldn't be loud enough in some case, but we're already supposed to have a `TableMetadata` instance. It's not like we'll just log at INFO if something higher up the stack passes a non-existent table name.